file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs

```diff
@@ -303,7 +345,75 @@ protected internal override async Task OnExecuteCommand()
+            if (result == MessageBoxResult.Yes)
+            {               
+                if (CancellationTokenSource != null && !CancellationTokenSource.IsCancellationRequested)
+                {
+                    logger.LogInformation("Cancel job confirmed for Action : {Action}, displayName:{displayName} and batch ID:{BatchId}.", Action, DisplayName, batchHandle?.BatchId);
+                    IsCanceled = true;
+                    IsCommitting = false;                                                               
+                    CancellationTokenSource.Cancel();                                        
+                } 
+            }
+            else
+            {
+                logger.LogInformation("Cancel job canceled for Action : {Action} and displayName :{displayName} and batch ID:{BatchId}.", Action, DisplayName, batchHandle?.BatchId);
+            }
+        }
+
+        private async Task<string> HandleCanceledOperationsAsync(IBatchHandle batchHandle, bool IsCommitting, IFileShareApiAdminClient fileShareClient, OperationCanceledException e, CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested && e.CancellationToken == cancellationToken)
+            {
+                logger.LogInformation("Cancel job execution started for Action : {Action}, displayName:{displayName} and batch ID:{BatchId}.", Action, DisplayName, batchHandle.BatchId);               
+                if (IsCommitting)
+                {
+                    logger.LogInformation("File Share Service SetExpiryDateAsync started for batch ID:{BatchId}.", batchHandle.BatchId);
+                    string expiryDateString = DateTime.UtcNow.AddDays(-7).ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture);
+                    await fileShareClient.SetExpiryDateAsync(batchHandle.BatchId, new BatchExpiryModel { ExpiryDate = expiryDateString }, CancellationToken.None);
+                    logger.LogInformation("File Share Service SetExpiryDateAsync completed for batch ID:{BatchId}.", batchHandle.BatchId);
+                }
+                else
+                {
+                    try
+                    {
+                        logger.LogInformation("File Share Service batch rollback started for batch ID:{BatchId}.", batchHandle.BatchId);
+                        await fileShareClient.RollBackBatchAsync(batchHandle);
+                        logger.LogInformation("File Share Service batch rollback completed for batch ID:{BatchId}.", batchHandle.BatchId);
+                    }
+                    catch (Exception ex)
+                    {
+                        if (((System.Net.Http.HttpRequestException)ex).StatusCode.Equals(HttpStatusCode.Conflict))
+                        {
+                            logger.LogInformation("File Share Service SetExpiryDateAsync started for batch ID:{BatchId}.", batchHandle.BatchId);
```

Comment:
Please can we update log message to include that rollback was attempted but got conflict and hence setting expiry.

Reply:
Thank you Vijendra for reviewing changes. Added log message as per review comment.